### PR TITLE
Add date-fns library

### LINKED ISSUE #8: Provide a "blessed" version of date/time functionality that doesn't require any dependencies.
Take a cut of the date-fns library and re-export.

### PR DESCRIPTION
**Pull request by carlansley**
Fixes #8

### CODE REVIEW

File: src/date-fns/format/index.ts [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+
+  const formatterOptions = {
+    firstWeekContainsDate: firstWeekContainsDate as FirstWeekContainsDate,
+    weekStartsOn: weekStartsOn as Day,
+    locale: locale,
+    _originalDate: originalDate,
+  };
+
+  const result = formatStr
+    .match(longFormattingTokensRegExp)!
```
[the remark this answers is not in the document]
Comment (carlansley) [reply src/date-fns/format/index.ts:377]: Issue with underlying date-fns library, but something to be aware of.

**Comment by carlansley** [comment]
Ignore everything under date-fns, its a slimmed down cut of the library.  It will be updated as new versions come out.

**Review by ramaghanta — APPROVED** [review]
lgtm

**Review by lukebrdn — APPROVED** [review]
lgtm

**Review by greg-w101 — APPROVED** [review]
lgtm

**Comment by carlansley** [comment]
I need to figure out why it's failing...

**Comment by carlansley** [comment]
So... it was failing because the `format` test was outputting in the local timezone, which was different between my Mac and the GitHub servers.  This led me to adding the `date-fns-tz` library, and I created a `formatUtc` helper function that does the right thing.  And is probably what we want to do in most cases, since we use UTC for almost everything.  Unfortunately there are a lot more changes, so re-requesting reviews...

**Review by ramaghanta — APPROVED** [review]
lgtm

**Review by bobwieler — APPROVED** [review]
lgtm

**Review by lukebrdn — APPROVED** [review]
lgtm

**Pull request merged**